genai_processors/contrib/openrouter_model.py

```diff
@@ -0,0 +1,424 @@
+
+  OpenRouter provides access to hundreds of AI models through a unified API,
+  including models from OpenAI, Anthropic, Google, Meta, and many others.
+  """
+
+  def __init__(
+      self,
+      *,
+      api_key: str,
+      model_name: str,
+      base_url: str | None = None,
+      site_url: str | None = None,
+      site_name: str | None = None,
+      generate_content_config: GenerateContentConfig | None = None,
+  ):
+    """Initialize the OpenRouter model.
+
+    Args:
+      api_key: Your OpenRouter API key.
+      model_name: Model to use (e.g., "openai/gpt-4o", "anthropic/claude-3-5-sonnet").
+      base_url: OpenRouter API base URL (defaults to https://openrouter.ai/api/v1).
+      site_url: Your site URL (optional, for OpenRouter rankings).
+      site_name: Your site name (optional, for OpenRouter rankings).
+      generate_content_config: Model configuration parameters.
+
+    Returns:
+      A `Processor` that calls the OpenRouter API with streaming support.
+    """
+    self._api_key = api_key
+    self._model_name = model_name
+    self._base_url = base_url or _DEFAULT_BASE_URL
+    self._site_url = site_url
+    self._site_name = site_name
+    self._config = generate_content_config or {}
+
+    # Build headers
+    headers = {
+        'Authorization': f'Bearer {self._api_key}',
+        'Content-Type': 'application/json',
+        'User-Agent': 'genai-processors',
+    }
+
+    if self._site_url:
+      headers['HTTP-Referer'] = self._site_url
+    if self._site_name:
+      headers['X-Title'] = self._site_name
+
+    self._client = httpx.AsyncClient(
+        base_url=self._base_url,
+        headers=headers,
+        timeout=_DEFAULT_TIMEOUT,
+    )
+
+  @property
+  def key_prefix(self) -> str:
+    """Key prefix for caching."""
+    return f'OpenRouterModel_{self._model_name}'
+
+  async def call(
+      self, content: AsyncIterable[content_api.ProcessorPartTypes]
+  ) -> AsyncIterable[content_api.ProcessorPartTypes]:
+    """Process content through OpenRouter API."""
+    messages = []
+    async for part in content:
+      if isinstance(part, content_api.ProcessorPart):
+        messages.append(_to_openrouter_message(part))
+      else:
+        # Convert other types to ProcessorPart first
+        processor_part = content_api.ProcessorPart(part)
+        messages.append(_to_openrouter_message(processor_part))
+
+    if not messages:
+      return
+
+    # Prepare request payload
+    payload = {
+        'model': self._model_name,
+        'messages': messages,
+        'stream': self._config.get('stream', True),
+    }
+
+    # Add configuration parameters
+    for key, value in self._config.items():
+      if key != 'stream' and value is not None:
+        payload[key] = value
+
+    try:
+      # Make streaming request
+      async with self._client.stream(
+          'POST',
+          '/chat/completions',
+          json=payload,
+      ) as response:
+        response.raise_for_status()
```

Comment:
raise_for_status by design does not include the error returned by the server. this makes triaging the error harder. In Ollama integration we did this:
http://google3/third_party/py/genai_processors/core/ollama_model.py;l=242;rcl=780667442
I don't know how OpenRouter returns errors, but maybe a similar handling could be added?